packages/shared/src/hooks/useJoinReferral.ts

```diff
@@ -0,0 +1,60 @@
+import { useEffect } from 'react';
+import request from 'graphql-request';
+import { useRouter } from 'next/router';
+import { setCookie } from '../lib/cookie';
+import { isDevelopment } from '../lib/constants';
+import { GET_REFERRING_USER_QUERY } from '../graphql/users';
+import { graphqlUrl } from '../lib/config';
+import { ApiErrorResult } from '../graphql/common';
+import { useAuthContext } from '../contexts/AuthContext';
+
+export const useJoinReferral = (): void => {
+  const { user } = useAuthContext();
+  const router = useRouter();
+  const { cid, userid } = router.query;
+
+  useEffect(() => {
+    if (user && userid === user.id) {
+      return;
+    }
+
+    const ONE_YEAR = 1 * 365 * 24 * 60 * 60;
+    const campaign = cid as string;
+    const referringUserId = userid as string;
+
+    if (campaign && referringUserId) {
+      setCookie('join_referral', `${referringUserId}:${campaign}`, {
+        maxAge: ONE_YEAR,
+        secure: !isDevelopment,
+        domain: process.env.NEXT_PUBLIC_DOMAIN,
+        sameSite: 'lax',
+      });
+
+      const removeCookie = () => {
```

Comment:
Should we not consider adding a removeCookie function in the cookie lib?
Could take a name for instance and do everything else inside?

Reply:
Added as `expireCookie` bb675b5c824b20bae4f2a26fd63c1ef9884b8d07